src/views/forms/AccountManagementFormsNew/WithdrawForm/NetworkSelectMenu.tsx

```diff
@@ -0,0 +1,156 @@
+import { Chain } from '@skip-go/client';
+import tw from 'twin.macro';
+
+import { cctpTokensByChainId, isHighFeeChainId, isLowFeeChainId } from '@/constants/cctp';
+import { SUPPORTED_COSMOS_CHAINS } from '@/constants/graz';
+import { STRING_KEYS } from '@/constants/localization';
+import { TransferType } from '@/constants/transfers';
+import { ConnectorType, WalletType } from '@/constants/wallets';
+
+import { useAccounts } from '@/hooks/useAccounts';
+import { useEnvFeatures } from '@/hooks/useEnvFeatures';
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { SearchSelectMenu } from '@/components/SearchSelectMenu';
+
+import { getSelectedDydxChainId } from '@/state/appSelectors';
+import { useAppSelector } from '@/state/appTypes';
+
+import { isTruthy } from '@/lib/isTruthy';
+
+import exchanges from '../../../../../public/configs/exchanges.json';
+import { FeeLevel, FeeLevelTag } from '../FeeLevelTag';
+
+type ElementProps = {
+  selectedExchange?: string;
+  selectedChain?: string;
+  onSelect: (name: string, type: 'chain' | 'exchange') => void;
+  chains: Chain[];
+};
+
+// TODO: fix this
+// our menu item types are wrong (has label typed as a react node even though we use string)
+// so we have to make our own type.
+type SelectableItem = {
+  value: string;
+  label: string;
+  onSelect: () => void;
+  slotBefore: JSX.Element;
+  slotAfter?: JSX.Element;
+};
+
+const solanaChainIdPrefix = 'solana';
+const DEFAULT_NETWORKS_LABEL = 'Networks';
+
+export const NetworkSelectMenu = ({
+  selectedExchange,
+  selectedChain,
+  onSelect,
+  chains,
+}: ElementProps) => {
+  const { sourceAccount } = useAccounts();
+  const selectedDydxChainId = useAppSelector(getSelectedDydxChainId);
+  const { CCTPWithdrawalOnly } = useEnvFeatures();
+
+  const stringGetter = useStringGetter();
+
+  const isKeplrWallet = sourceAccount.walletInfo?.name === WalletType.Keplr;
+
+  const getFeeDecoratorComponentForChainId = (chainId: string) => {
+    if (isLowFeeChainId(chainId, TransferType.Withdraw))
+      return <FeeLevelTag feeLevel={FeeLevel.Low} />;
+    if (isHighFeeChainId(chainId, TransferType.Withdraw))
+      return <FeeLevelTag feeLevel={FeeLevel.High} />;
+    return undefined;
+  };
+
+  const chainItems = chains
+    .map((chain) => ({
+      value: chain.chainID,
+      label: chain.chainName,
+      onSelect: () => {
+        onSelect(chain.chainID, 'chain');
+      },
+      slotBefore: <$Img src={chain.logoURI ?? undefined} alt="" />,
+      slotAfter: getFeeDecoratorComponentForChainId(chain.chainID),
+    }))
+    .filter((chain) => {
+      // only cosmos chains are supported on kepler
+      if (isKeplrWallet) {
```

Comment:
for withdrawing, why does the connected wallet matter? cant i choose to withdraw to a cosmos chain even when connected to metamask, since it doesn't require signature from my source wallet?